### DISCUSSION TITLE
Share Your High Availability Architecture thread

### DISCUSSION THREAD

**Discussion by NorseGaud** [docker-mailserver/docker-mailserver#2391] [Show and tell]
I'd like to start this thread and allow users of the docker-mailserver to explain a bit about how they achieve High Availability when using docker-mailserver.

First, definition of High Availability: 

> High availability (HA) is a component of a technology system that eliminates single points of failure to ensure continuous operations or uptime for an extended period. High Availability ensures your systems, databases, and applications operate when and as needed.

---

I'll start off by sharing how I achieve HA:

1. 2 x t2.micro AWS instances, one in Europe and one in the US, paid for annually so they're cheap (I decided against ECS here due to the cost and lack of flexibility/troubleshooting access)
2. Cross region EFS volume to store the docker-mailserver data directory
3. EFS backed up with [Amazon Backup](https://docs.aws.amazon.com/efs/latest/ug/awsbackup.html)
2. Route53 setup on the DNS name I use + [DNS failover](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/dns-failover-configuring.html) so status changes for the instances will cause the domain's IP to change (low TTL) to the healthy instance temporarily.

I decided to stop here because it met my needs, but my goal was to eventually have another US based instance.

**Comment by TimothyGaray** [comment]
So far I am still doing good with just a [free] function app in Azure monitoring the responsiveness of the SMTP server at home.  Still tweaking the script to eliminate false alarms but at least I know it is running.
